SingleSource/UnitTests/HashRecognize/crc_macros.h

```diff
@@ -0,0 +1,64 @@
+#define SMIN(ty)                                                               \
+  _Generic(((ty)0), uint8_t: 0x80, uint16_t: 0x8000, uint32_t: 0x80000000)
+
+#define BW(ty) _Generic(((ty)0), uint8_t: 8, uint16_t: 16, uint32_t: 32)
```

Comment:
```suggestion
#define BW(ty) (sizeof(ty) * 8)

#define SMIN(ty) (1 << BW(ty) - 1)
```

And this probably avoids the need for C11.